lib/adal-angular.js

```diff
@@ -83,11 +83,8 @@
                         var requestInfo = _adal.getRequestInfo(hash);
                         _adal.saveTokenFromHash(requestInfo);
 
-                        if ($location.$$html5) {
-                            $window.location = $window.location.origin + $window.location.pathname;
-                        } else {
-                            $window.location.hash = '';
-                        }
+                        // resetting hash to remove fragments sent by AAD
+                        $location.hash('');
```

Comment:
@weijjia $location.hash('') will not trigger another location change. Updating $window.location was doing that.